apps/examples/testcase/le_tc/kernel/tc_libc_misc.c

```diff
@@ -168,6 +178,182 @@ static void tc_libc_misc_crc16part(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+ * @fn                  :tc_libc_misc_crc16_ccitt_xmodem
+ * @brief               :Return a 16-bit CRC16-CCITT/Xmodem of the contents of the 'src' buffer, length 'len'
+ * @scenario            :Return a 16-bit CRC16-CCITT/Xmodem of the contents of the 'src' buffer, length 'len'
+ * @API's covered       :crc16_ccitt_xmodem
+ * @Preconditions       :None
+ * @Postconditions      :None
+ * @Return              :void
+ */
+static void tc_libc_misc_crc16_ccitt_xmodem(void)
+{
+	uint16_t ret_chk;
+	uint8_t src_arr[1] = { VAL_100 };
+	size_t length = 1;
+
+	/* Return value should be 11298 as calculated by crc16_ccitt_xmodem */
+
+	ret_chk = crc16_ccitt_xmodem(src_arr, length);
+	TC_ASSERT_EQ("crc16_ccitt_xmodem", ret_chk, VAL_11298);
+
+	/* Return value should be 5649 as calculated by crc16_ccitt_xmodem */
+
+	src_arr[0] = VAL_50;
+	ret_chk = crc16_ccitt_xmodem(src_arr, length);
+	TC_ASSERT_EQ("crc16_ccitt_xmodem", ret_chk, VAL_5649);
+
+	TC_SUCCESS_RESULT();
+}
```

Comment:
How about below?
```
#define ARR_NUM 2
#define IN_VAL_100 100
#define IN_VAL_50 50
#define CRC_RES_OF_100 11298  // result value of crc16_ccitt_xmodem for input 100
#define CRC_RES_OF_50 5649 // result value of crc16_ccitt_xmodem for input 50

static void tc_libc_misc_crc16_ccitt_xmodem(void)
{
	uint16_t ret_chk;
	uint8_t src_arr[2] = { IN_VAL_100, IN_VAL_50 };
	uint8_t res_arr[2] = { CRC_RES_OF_100, CRC_RES_OF_50 };
	size_t length = 1;

	/* check crc16_ccitt_xmodem results according to input values */

	for (int arr_idx = 0; arr_idx < ARR_NUM; arr_idx++) {
		ret_chk = crc16_ccitt_xmodem(&src_arr[arr_idx], length);
		TC_ASSERT_EQ("crc16_ccitt_xmodem", ret_chk, res_arr[arr_idx]);
	}

	TC_SUCCESS_RESULT();
}
```

Reply:
I think it's good idea to make array for src and res.
But it seems better to leave the definition variable as it is, since there are overlapping numbers among the result values.